### PR TITLE
Improve `opensearch.pluginzip` docs.

### PR DESCRIPTION
**Pull request by prudhvigodithi**
Signed-off-by: pgodithi <pgodithi@amazon.com>

### Description
Improve `opensearch.pluginzip` docs, to add more details, purpose and boundaries of usage.

### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/3692

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -173,18 +173,18 @@ Use `./gradlew publishShadowPublicationToStagingRepository` to produce maven art
 
 ### opensearch.pluginzip
 
-This plugin identifies the generated zip file distribution which is the ouput of `bundlePlugin` task and publishes to local maven repo with standardized maven coordinates.
+`opensearch.pluginzip` is designed to facilitate OpenSearch plugin zips to be available in maven repo which can then be fetched for dependency using maven coordinates. This plugin identifies the OpenSearch plugin zip file, the ouput of `bundlePlugin` task and publishes to local maven repo with standard maven coordinates.
```
Comment (reta) [inline BUILDING.md:176]:
```suggestion
`opensearch.pluginzip` is designed to facilitate OpenSearch plugin zips to be available in maven repo which can then be fetched for dependency using Apache Maven repositories. This plugin identifies the OpenSearch plugin ZIP file, the output of `bundlePlugin` task and publishes to local Apache Maven repository.
```

File: BUILDING.md [hunk 2]
```diff
@@ -173,18 +173,18 @@ Use `./gradlew publishShadowPublicationToStagingRepository` to produce maven art
...
 
 #### Plugin Design
 
 * `opensearch.pluginzip` is java based code published into build-tools.
 * The maven coordinates groupID is fixed as `org.openserach.plugin`, `version` and `artifcatID` will be inferred from gradle project properties.
-* User can also pass custom POM extensions, that will add desired xml to the zip maven POM file generated during runtime.
-* Once the plugin is added to the `build.gradle` as `apply plugin: 'opensearch.pluginzip'`, this will add a new custom publish task `publishPluginZipPublicationToZipStagingRepository`, this task will publish the zip distribution to the maven local (file system).
-* The maven local artifacts can then be published to maven central/nexus.
-* The plugin will not add jars generated by tasks `sourcesJar` and `javadocJar`, this is done in purpose to exclude `jars` for `zip` publications.
+* User can also pass custom POM extensions, that will add desired properties to maven POM file generated during runtime.
```
Comment (reta) [inline BUILDING.md:184]:
```suggestion
* User can also pass custom POM extensions, that will add desired properties to Apache Maven POM file generated during runtime.
```

File: BUILDING.md [hunk 3]
```diff
@@ -173,18 +173,18 @@ Use `./gradlew publishShadowPublicationToStagingRepository` to produce maven art
...
+* Once the plugin is added to the `build.gradle` as `apply plugin: 'opensearch.pluginzip'`, this will add a new custom publish task `publishPluginZipPublicationToZipStagingRepository`, the purpose of this task is to publish the zip distribution to the maven local (file system).
```
Comment (reta) [inline BUILDING.md:185]:
```suggestion
* Once the plugin is added to the `build.gradle` as `apply plugin: 'opensearch.pluginzip'`, this will add a new custom publish task `publishPluginZipPublicationToZipStagingRepository`, the purpose of this task is to publish the ZIP distribution to the Apache Maven local repository (file system).
```

File: BUILDING.md [hunk 4]
```diff
@@ -173,18 +173,18 @@ Use `./gradlew publishShadowPublicationToStagingRepository` to produce maven art
...
+* The maven local artifacts will then be published to maven central/nexus using [CI workflows](https://github.com/opensearch-project/opensearch-build/tree/main/jenkins).
```
Comment (reta) [inline BUILDING.md:186]:
```suggestion
* The Apache Maven local artifacts could then be published to  Apache Maven central/nexus using [CI workflows](https://github.com/opensearch-project/opensearch-build/tree/main/jenkins).
```

File: BUILDING.md [hunk 5]
```diff
@@ -173,18 +173,18 @@ Use `./gradlew publishShadowPublicationToStagingRepository` to produce maven art
...
+* The plugin will not add generated jars by tasks `sourcesJar` and `javadocJar`, this is done in purpose to exclude `jars` for `zip` publications.
```
Comment (reta) [inline BUILDING.md:187]:
```suggestion
* The plugin will not publish generated JARs (`sourcesJar`, `javadocJar`), this is done on purpose to separate `jars` and `zip` publications.
```

**Review by reta — DISMISSED** [review]

**Review by gaiksaya — DISMISSED** [review]

File: BUILDING.md [hunk 6]
```diff
@@ -210,10 +210,10 @@ buildscript {
 3. Add `apply plugin: 'opensearch.pluginzip'` to the build.gradle file.
 Once added, this should list the new task `publishPluginZipPublicationToZipStagingRepository`.
 
-4. Run the task `publishPluginZipPublicationToZipStagingRepository` (add it to managed build script build.sh)  
+4. Run the task `publishPluginZipPublicationToZipStagingRepository` (add it to managed build script `build.sh`)  
 ```./gradlew publishPluginZipPublicationToZipStagingRepository -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT -Dbuild.version_qualifier=$QUALIFIER```
 
-    Note: The gradle assemble task `./gradlew assemble` should be called first before calling `publishPluginZipPublicationToZipStagingRepository`, as zip file need to be generated first before publishing.
+    Note: The gradle assemble task `./gradlew assemble` should be called first before calling `publishPluginZipPublicationToZipStagingRepository`, as zip file need to be generated before publishing.
```
Comment (saratvemulapalli) [inline BUILDING.md:216]: Not sure if plugins support `./gradle assemble`, i know few plugins use `./gradle build`[1]

[1] https://github.com/opensearch-project/anomaly-detection/blob/main/DEVELOPER_GUIDE.md#building-from-the-command-line

**Review by saratvemulapalli — APPROVED** [review]

File: BUILDING.md (hunk 6 above)
Comment (prudhvigodithi) [reply BUILDING.md:216]: Thanks for the inputs @saratvemulapalli and @ylwu-amzn, the idea to mention the zip has be available (either by calling `assemble` or `build`) before publishing, let me rephrase this way and let me know this good to go with.

Note: The gradle task responsible to generate the distribution zip should be called first before executing `publishPluginZipPublicationToZipStagingRepository`, as the zip file needs to be generated before publishing.

**Review by saratvemulapalli — APPROVED** [review]

**Review by gaiksaya — APPROVED** [review]

**Review by bbarani — APPROVED** [review]
